hs_composite_resource/models.py

```diff
@@ -0,0 +1,125 @@
+import os
+
+from django.core.exceptions import ValidationError
+
+from mezzanine.pages.page_processors import processor_for
+
+from hs_core.models import BaseResource, ResourceManager, resource_processor
+
+
+class CompositeResource(BaseResource):
+    objects = ResourceManager("CompositeResource")
+
+    class Meta:
+        verbose_name = 'Composite'
+        proxy = True
+
+    @property
+    def can_be_public_or_discoverable(self):
+        if not super(CompositeResource, self).can_be_public_or_discoverable:
+            return False
+        for lf in self.logical_files:
```

Comment:
The logic is based on how we have currently implemented various resource types. If a specific resource type is missing some required resource specific metadata, we don't allow it to be discoverable. So if RasteLogicalFile type is missing any required raster specific metadata then I think the resource (composite) should not be discoverable. However, if want to apply different logic for composite resource to be discoverable that change can be made.